File: locations/kitchen.pl
:- multifile describe/1, dwight_stapler_out_clear/0, dwight_stapler_out_jelly/0.


describe(kitchen) :-
    write("You are in the kitchen"), nl,
    write("You can go to the 'break_room', 'bathroom' or to the 'openspace' from here"), nl,
    write("There is a coffee machine, some tea, kettle, and jelly powder."),
    write("Maybe it will be useful later..."),

    (dwight_stapler_out_clear ->
        nl,
        write("Do you want to put the Dwight stapler to the jelly?"), nl,
        write("HINT: 'jelly_stapler.'")
        ;
        true
    ).


jelly_stapler :-
    i_am_at(kitchen),
    dwight_stapler_out_clear,

    write("Choose the color of the jelly."), nl,
    write("HINT: 'red', 'green' or 'blue': "), nl,
    read(_),

    write("Its ready!"), nl, nl,
    write(" /----------\\"), nl,
    write("/  |/----,,  \\"), nl,
    write("|  |[______   |"), nl,
    write("==============="), nl, nl,

    write("Now I can put back the stapler to the drawer."), nl,

    retractall(dwight_stapler_out_clear),
    assert(dwight_stapler_out_jelly),
    retractall(holding(stapler)),
    assert(holding(stapler_in_jelly)).

File: Haskell/Kitchen.hs
module Kitchen where

import Game
import Rooms
import RemoveItem

describe :: JimQuest -> IO ()
describe jimQuest = do
  putStrLn "You are in the kitchen"
  putStrLn "There is a coffee machine, some tea, a kettle, and some jelly powder."
  putStrLn "Maybe it will be useful later..."
  putStrLn "You can go to the 'break_room', 'bathroom' or to the 'openspace' from here."
  if jimQuest == StaplerOutClear
    then do
      putStrLn "Do you want to put the Dwight stapler to the jelly?"
      putStrLn "HINT: jelly_stapler"
    else return ()


jellyStapler :: Game -> IO Game
jellyStapler (Game iamAt saidHi jimQuest creedQuest dwightQuest inventory) = do
    if ((iamAt /= Kitchen) || (jimQuest /= StaplerOutClear))
    then do
        return (Game iamAt saidHi jimQuest creedQuest dwightQuest inventory)
    else do
      putStrLn ""
      putStrLn "It's ready!"
      putStrLn ""
      putStrLn " /----------\\"
      putStrLn "/  |/----,,  \\"
      putStrLn "|  |[______   |"
      putStrLn "==============="
      putStrLn "Now I can put back the stapler to the drawer."
      putStrLn ""
      let itemsWithStaplerInJelly = inventory ++ [StaplerInJellyItem]
      let itemsWithoutStapler = removeItem itemsWithStaplerInJelly Stapler
      return (Game iamAt saidHi StaplerOutJelly creedQuest dwightQuest itemsWithoutStapler)
